### ISSUE TITLE
SingleNodeExecutor hanging forever

### ISSUE THREAD

**Issue by anatolysergeev** [zio/zio-redis#674]
I simulated redis node failover and this shows that no errors returns and send commands waits answer forever
SingleNodeExecutor doesn't throw connection errors, timeout errors, it is hanging forever

**Comment by kuzkdmy** [comment]
Got the same issue

**Comment by mijicd** [comment]
@kuzkdmy Working on a fix, thanks for reporting it :pray:

**Comment by mijicd** [comment]
No, still open, also open for "grabs". Work I referred to above was an overhaul of lib's internals, which should've probably be allocated to another issue as it's a big chunk of work (which is why it's still not complete nor actively worked on 🤦 ).
